testing/hyperparams/test_space.py

```diff
@@ -89,3 +90,51 @@ def test_hyperparams_space_rvs_outputs_samples():
     for k, v in samples.iter_flat():
         assert k in space
         assert not isinstance(v, HyperparameterDistribution)
+
+
+def test_hyperparameter_samples_compress():
+    hps = HyperparameterSamples({
+        "b__a__learning_rate": 7,
+        "b__learning_rate": 9,
+        "Sklearn__test1__test__abc": False,
+        "Sklearn__test2__test__abc": "Parallel"
+    }, separator="__")
+    expected = [
+        {'step_name': 'a', 'hyperparams': {'learning_rate': 7}, 'ancestor_steps': ['b']},
+        {'step_name': 'b', 'hyperparams': {'learning_rate': 9}, 'ancestor_steps': []},
+        {'step_name': 'test', 'hyperparams': {'abc': False}, 'ancestor_steps': ["Sklearn", "test1"]},
+        {'step_name': 'test', 'hyperparams': {'abc': "Parallel"}, 'ancestor_steps': ["Sklearn", "test2"]}]
+
+    actual = hps.compress()
+    assert isinstance(actual, CompressedHyperparameterSamples)
+    assert str(actual) == str(expected)
+
+
+def test_hyperparameter_samples_compress_without_parents():
+    hps = HyperparameterSamples({
+        "b__a__learning_rate": 7,
+        "b__learning_rate": 9,
+        "Sklearn__test1__test__abc": False,
+        "Sklearn__test2__test__abc": "Parallel"
+    }, separator="__")
+    expected = [
+        {'step_name': 'a', 'hyperparams': {'learning_rate': 7}, 'ancestor_steps': None},
+        {'step_name': 'b', 'hyperparams': {'learning_rate': 9}, 'ancestor_steps': None},
+        {'step_name': 'test', 'hyperparams': {'abc': False}, 'ancestor_steps': None},
+        {'step_name': 'test', 'hyperparams': {'abc': "Parallel"}, 'ancestor_steps': None}]
+    actual = hps.compress(remove_parents=True)
+    assert isinstance(actual, CompressedHyperparameterSamples)
+    assert str(actual) == str(expected)
+
+
+def test_hyperparameter_samples_compress_wildcards():
```

Comment:
I suggest using one simple parametrized test for all the scenarios suggested by Vincent. 

Example of parametrized tests: 
https://github.com/Neuraxio/Neuraxle/search?q=%40pytest.mark.parametrize%28

The actuals and expecteds could be sent as arguments within lists of arguments to the parametrized test. That would be the single best way to test the different cases, instead of doing a separate test for each. That reduces code duplication, and makes it simpler to test and maintain. 

In the parametrized tests listed in the above GitHub code search link, the following one particularily reflects how it would be easy to parametrize the test: 
https://github.com/Neuraxio/Neuraxle/blob/3091bbc175be33b3b415386cf8a4dc79ce48d368/testing/steps/test_features.py

With the difference that the expected variable here would not be a single integer of feature counts, but instead an `expected_items` dict items of the wildcarded things.